### ISSUE TITLE
Provider authentication is broken in v0.12.0-0

### ISSUE THREAD

**Issue by oponomarov-tu** [OpsLevel/terraform-provider-opslevel#303]
1. Using `api_token` in `opslevel` provider configuration stopped working, even if we hardcode it in code. Worked like a charm in [v0.11.0](https://github.com/OpsLevel/terraform-provider-opslevel/releases/tag/v0.11.0)
2. New version reports warning about `OPSLEVEL_API_TIMEOUT` which we do not set anywhere and rely on provider defaults. This shouldn't be here.

```terraform plan
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Warning: Expected OPSLEVEL_API_TIMEOUT to be an int
│
│   with provider["registry.terraform.io/opslevel/opslevel"],
│   on main.tf line 66, in provider "opslevel":
│   66: provider "opslevel" {
│
│ OPSLEVEL_API_TIMEOUT was set but not as an int. The default timeout value of 30 seconds will be used.
╵
╷
│ Error: Missing OPSLEVEL_API_TOKEN
│
│   with provider["registry.terraform.io/opslevel/opslevel"],
│   on main.tf line 66, in provider "opslevel":
│   66: provider "opslevel" {
│
│ An OPSLEVEL_API_TOKEN is needed to authenticate with the opslevel client. This can be set as an environment variable or in the provider configuration block as 'api_token'.

**Comment by rocktavious** [comment]
Thanks for the report @oponomarov-tu - the 0.12 pre release is a complete rewrite under the hood using the latest terraform SDK.  Looks like we missed this so we'll get this fixed in the next pre-release we expect to ship on April 26th

**Comment by rocktavious** [comment]
Fixed in v0.13.0-1 pre-release
